test/unit/decoded_view.cpp

```diff
@@ -0,0 +1,508 @@
+            decoded_view s2("a%20vri+test");
+
+            string_view str0(dec_str);
+            string_view str1("a tri test");
+            string_view str2("a vri test");
+
+            std::string bstr0(dec_str);
+            std::string bstr1("a tri test");
+            std::string bstr2("a vri test");
+
+            BOOST_TEST(s == s0);
+            BOOST_TEST_NOT(s == s1);
+            BOOST_TEST(s == str0);
+            BOOST_TEST_NOT(s == str1);
+            BOOST_TEST(s == bstr0);
+            BOOST_TEST_NOT(s == bstr1);
+
+            BOOST_TEST(s != s2);
+            BOOST_TEST_NOT(s != s0);
+            BOOST_TEST(s != str2);
+            BOOST_TEST_NOT(s != str0);
+            BOOST_TEST(s != bstr2);
+            BOOST_TEST_NOT(s != bstr0);
+
+            BOOST_TEST(s < s2);
+            BOOST_TEST_NOT(s < s0);
+            BOOST_TEST(s < str2);
+            BOOST_TEST_NOT(s < str0);
+            BOOST_TEST(s < bstr2);
+            BOOST_TEST_NOT(s < bstr0);
+
+            BOOST_TEST(s <= s2);
+            BOOST_TEST(s <= s0);
+            BOOST_TEST(s <= str2);
+            BOOST_TEST(s <= str0);
+            BOOST_TEST(s <= bstr2);
+            BOOST_TEST(s <= bstr0);
+
+            BOOST_TEST(s > s1);
+            BOOST_TEST_NOT(s > s0);
+            BOOST_TEST(s > str1);
+            BOOST_TEST_NOT(s > str0);
+            BOOST_TEST(s > bstr1);
+            BOOST_TEST_NOT(s > bstr0);
+
+            BOOST_TEST(s >= s1);
+            BOOST_TEST(s >= s0);
+            BOOST_TEST(s >= str1);
+            BOOST_TEST(s >= str0);
+            BOOST_TEST(s >= bstr1);
+            BOOST_TEST(s >= bstr0);
+
+        }
+    }
+
+    void
+    testConversion()
+    {
+        // to_string()
+        {
+            decoded_view s(str);
+            std::string o = "to be recycled";
+            BOOST_TEST_EQ(s.to_string(o), dec_str);
+        }
+
+        // to_string()
+        {
+            decoded_view s(str);
+            BOOST_TEST_EQ(s.to_string(), dec_str);
+        }
+
+        // explicit operator basic_string()
+        {
+            decoded_view s(str);
+            std::string o(s);
+            BOOST_TEST_EQ(o, dec_str);
+        }
+
+        // pass it to a function taking a string_view
+        {
+            auto fn = [this](string_view sv)
+            {
+                BOOST_TEST(sv == dec_str);
+            };
+            decoded_view s(str);
+            fn(s.to_string());
+        }
+
+        // pass it to a function taking a char*
+        {
+            auto fn = [this](char const* sv)
+            {
+                BOOST_TEST(sv == dec_str);
+            };
+            decoded_view s(str);
+            fn(s.to_string().c_str());
+        }
+    }
+
+    void
+    testAppend()
+    {
+        // to_string()
+        {
+            decoded_view s(str);
+            std::string o = "init ";
+            s.append_to(o);
+            std::string exp = std::string("init ");
+            exp += dec_str;
```

Comment:
so we are converting `dec_str` to a basic_string implicitly? oof... if there's a big SQL expression that's a lot of temporary allocations.

Reply:
I'm not sure I understand the question, but `exp` is the expected string result, and I'm just using `dec_str` to assemble the expected results. This is not touching and has nothing to do with `decoded_view` at all.

Reply:
No. 

```cpp
string_view dec_str = "a uri test";
```

It's just a fixture with the expected result of that test. 

It has nothing to do with what users will do with a decoded_view.